librz/il/definitions/float.c

```diff
@@ -0,0 +1,398 @@
+#include <rz_il/definitions/float.h>
+
+/** For IEEE 754 only **/
+/** 32/64/128 bits **/
+/** check : http://weitz.de/ieee/ **/
+/** check : https://www.h-schmidt.net/FloatConverter/IEEE754.html **/
+/**
+ * Interprets `x` as a floating number
+ * \param r Float format, an interpretation to bitvector x
+ * \param bv BitVector x
+ * \return f RzILFloat
+ */
+
+static ut32 get_float_format_len(RzILFloatFormat r) {
+	switch (r) {
+	case RZIL_FLOAT_IEEE754_32:
+		return 32;
+	case RZIL_FLOAT_IEEE754_64:
+		return 64;
+	case RZIL_FLOAT_IEEE754_128:
+		return 128;
+	default:
+		rz_warn_if_reached();
+		return 0;
+	}
+}
+
+ut32 rzil_float_get_bias_num(RZ_NONNULL RzILFloat *f) {
+	rz_return_val_if_fail(f, 0);
+	switch (f->r) {
+	case RZIL_FLOAT_IEEE754_32:
+		return 127;
+	case RZIL_FLOAT_IEEE754_64:
+		return 1023;
+	case RZIL_FLOAT_IEEE754_128:
+		return 16383;
+	default:
+		rz_warn_if_reached();
+		return 0;
+	}
+}
+
+ut32 rzil_float_get_exp_start_pos(RZ_NONNULL RzILFloat *f) {
+	rz_return_val_if_fail(f, 0);
+	switch (f->r) {
+	case RZIL_FLOAT_IEEE754_32:
+		return 23;
+	case RZIL_FLOAT_IEEE754_64:
+		return 52;
+	case RZIL_FLOAT_IEEE754_128:
+		return 112;
+	default:
+		rz_warn_if_reached();
+		return 0;
+	}
+}
+
+static bool check_valid_float_bv(RzILFloat *f) {
+	rz_return_val_if_fail(f, false);
+	ut32 len = rz_il_bv_len(f->s);
+	ut32 valid_len = get_float_format_len(f->r);
+	return len == valid_len;
+}
+
+RZ_API ut32 rzil_float_get_exp_len(RZ_NONNULL RzILFloat *f) {
+	rz_return_val_if_fail(f, 0);
+	ut32 exp_len = get_float_format_len(f->r) - rzil_float_get_exp_start_pos(f) - 1;
+	return exp_len;
+}
+
+RZ_API ut32 rzil_float_get_frac_len(RZ_NONNULL RzILFloat *f) {
+	rz_return_val_if_fail(f, 0);
+	ut32 exp_len = rzil_float_get_exp_len(f);
+	ut32 frac_len = get_float_format_len(f->r) - exp_len - 1;
+	return frac_len;
+}
+
+RZ_API RzILFloat *rzil_float_new(RzILFloatFormat r, RzILBitVector *bv) {
+	RzILFloat *result = RZ_NEW0(RzILFloat);
+	if (!result || !bv) {
+		return NULL;
```

Comment:
remove `!bv` and add the `rz_return_val_if_fail` before checking `r` and `bv`.

Reply:
> @Heersin how is it going? Were there any updates since? Could you please also rebase on top of the latest `dev` and solve the conflicts?



> @Heersin how is it going? Were there any updates since? Could you please also rebase on top of the latest `dev` and solve the conflicts?

Recently working on float test basic arithmetic cases and debugging (https://github.com/Heersin/rizin/tree/rz_util_float). I will continue it next week